### PR TITLE
add Block.average_daily_users property

### LINKED ISSUE #14407: Block refactoring - add .active_daily_user and drop FakeAddon
splitting out some of the work from #14391

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #14407

### CODE REVIEW

File: src/olympia/blocklist/models.py [hunk 1]
```diff
@@ -361,9 +362,11 @@ def save(self, *args, **kwargs):
     @classmethod
     def _split_guids_full(cls, guids, v_min, v_max, filter_existing):
         # load all the Addon instances together
-        addon_qs = Addon.unfiltered.filter(guid__in=guids).order_by(
-            '-average_daily_users').only_translations()
-        addons = list(addon_qs)
+        addons_qs = Addon.unfiltered.filter(guid__in=guids).only_translations()
+        addons = sorted(
```
Comment (bobsilverberg) [inline src/olympia/blocklist/models.py:366]: How are these new lines different from what the old lines were doing?

File: src/olympia/blocklist/models.py [hunk 2]
```diff
@@ -409,14 +412,15 @@ def _split_guids_full(cls, guids, v_min, v_max, filter_existing):
...
-            cls.FakeAddon(guid, addon_users) for guid, addon_users in addon_qs]
-        addon_guid_dict = {addon.guid: addon for addon in addons}
+        addons_qs = Addon.unfiltered.filter(guid__in=guids).values_list(
+            'guid', 'average_daily_users', named=True)
+
+        addons = sorted(
+            addons_qs,
+            key=lambda addon: addon.average_daily_users,
+            reverse=True)
+        adu_lookup = {
```
Comment (bobsilverberg) [inline src/olympia/blocklist/models.py:422]: Same question as above. How is this new code different from what was there before?

File: src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html [hunk 3]
```diff
@@ -5,10 +5,8 @@ <h3>{% blocktrans with count=blocks|length %}{{ count }} Add-on GUIDs{% endblock
         {% for block_obj in blocks %}
           <li>
             {{ block_obj.guid }}.
-            {% if block_obj.addon %}
-              <span class="addon-name">{{ block_obj.addon.name }}</span>
-              ({{ block_obj.addon.average_daily_users }} {% trans 'users' %}).
-            {% endif %}
+            <span class="addon-name">{{ block_obj.addon.name }}</span>
```
Comment (bobsilverberg) [inline src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html:8]: Why did we need an `if` statement to check for `block_obj.addon` before, but no longer need it? We're still accessing `block_obj.addon.name`.

File: src/olympia/blocklist/models.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/models.py:366]: they're the same - the follow on patch does it this way and I brought it forward (or left it in, depending on how you think about it)

File: src/olympia/blocklist/models.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/blocklist/models.py:422]: same as above.  The `adu_lookup` is a change though

File: src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html (hunk 3 above)
Comment (eviljeff) [reply src/olympia/blocklist/templates/admin/blocklist/includes/enhanced_blocks.html:8]: `block_obj.addon` won't exist now when the `block_obj` is a `FakeBlock` (i.e. when a large number of blocks are loaded).  Template parsing just ends up resolving missing attributes to `''` so it was in an `if` so it didn't look like ` users` when addon didn't exist.

**Comment by eviljeff** [comment]
> I am also wondering, why are there no changes to tests? Is this just a 100% refactoring task? Are there no tests that care that the shape of FakeBlock has changed?

Yeah, it's all refactoring so make the followup patch less noisy.  That no tests need changing is a good thing :)

**Pull request merged**